exchanges/okx/okx_wrapper.go

```diff
@@ -1446,3 +1430,29 @@ func (ok *Okx) GetHistoricCandlesExtended(ctx context.Context, pair currency.Pai
 	ret.SortCandlesByTimestamp(false)
 	return ret, nil
 }
+
+// GetAvailableTransferChains returns the available transfer blockchains for the specific
+// cryptocurrency
+func (ok *Okx) GetAvailableTransferChains(ctx context.Context, cryptocurrency currency.Code) ([]string, error) {
+	currencyChains, err := ok.GetFundingCurrencies(ctx)
+	if err != nil {
+		return nil, err
+	}
+	chains := make([]string, 0, len(currencyChains))
+	for x := range currencyChains {
+		if ((!cryptocurrency.IsEmpty() && strings.EqualFold(cryptocurrency.String(), currencyChains[x].Currency)) || cryptocurrency.String() == "") && currencyChains[x].Chain != "" {
+			chains = append(chains, currencyChains[x].Chain)
+		} else if currencyChains[x].MainNet && (!cryptocurrency.IsEmpty() && strings.EqualFold(cryptocurrency.String(), currencyChains[x].Currency)) {
+			depositAddress, err := ok.GetCurrencyDepositAddress(context.Background(), currencyChains[x].Currency)
+			if err != nil {
+				return nil, err
+			}
+			for y := range depositAddress {
+				if depositAddress[y].Selected && currencyChains[x].MainNet {
+					chains = append(chains, depositAddress[y].Chain)
+				}
+			}
+		}
```

Comment:
```go
for x := range currencyChains {
		if !cryptocurrency.IsEmpty() && !strings.EqualFold(cryptocurrency.String(), currencyChains[x].Currency) {
			continue
		}
		chains = append(chains, currencyChains[x].Chain)
	}
```

Pretty much achieves the same result. Concerning check `currencyChains[x].Chain != ""` after calling GetFundingCurrencies there is no Chain field that has not been set. So the resulting if statements never trigger and GetCurrencyDepositAddress function call never gets hit.